### PR TITLE
test: add api and doc http test

### PR DESCRIPTION
**Pull request by jun0315**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

add http's api and doc IT.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)

https://github.com/GreptimeTeam/greptimedb/issues/983

### CODE REVIEW

**Comment by shuiyisong** [comment]
iirc we already have a `DummyInstance`, could we reuse that?

**Comment by waynexia** [comment]
Not sure if [integration-test](https://github.com/GreptimeTeam/greptimedb/blob/afac885c1089df62701625fd49bc05c0dae299f3/tests-integration/tests/http.rs) is a better place for this kind of URL test. cc @sunng87

**Comment by sunng87** [comment]
@waynexia Since these test only covers http server I think it's ok to keep it here

File: src/servers/tests/http/http_test.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    async fn do_query(&self, _: &str, _: QueryContextRef) -> Vec<Result<Output>> {
+        unimplemented!()
+    }
+
+    async fn do_promql_query(
+        &self,
+        _: &str,
+        _: QueryContextRef,
+    ) -> Vec<std::result::Result<Output, Self::Error>> {
```
Comment (Fengys123) [inline src/servers/tests/http/http_test.rs:41]:
```suggestion
    ) -> Vec<Result<Output>> {
```

**Review by Fengys123 — APPROVED** [review]
LGTM

**Review by shuiyisong — APPROVED** [review]
LGTM

**Pull request merged**